sklearn/utils/testing.py

```diff
@@ -882,3 +882,128 @@ def check_docstring_parameters(func, doc=None, ignore=None, class_name=None):
             if n1 != n2:
                 incorrect += [func_name + ' ' + n1 + ' != ' + n2]
     return incorrect
+
+
+def check_data(doc_list, type_dict, type_name, object_name, include, exclude):
+    for name, type_definition, description in doc_list:
+        # remove all whitespaces
+        type_definition = type_definition.replace(' ', '')
+        description = [element.replace(' ', '') for element in description]
+
+        if name in type_dict:
+            u_dict = type_dict[name]
+            if (u_dict['type_definition'] != type_definition or
+                    u_dict['description'] != description):
+                if exclude is None:
+                    if (include == '*' or name in include):
+                        raise AssertionError(type_name+" "+name+" of " +
+                                             object_name+" has inconsistency.")
+                else:
+                    if name not in exclude:
+                        raise AssertionError(type_name+" "+name+" of " +
+                                             object_name+" has inconsistency.")
+        else:
+            if include is None:
+                if name not in exclude:
+                    add_dict = {}
+                    add_dict['type_definition'] = type_definition
+                    add_dict['description'] = description
+                    type_dict[name] = add_dict
```

Comment:
This would be more readable if you just defined add_dict here rather than a series of insertions